src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -582,15 +610,30 @@ def is_smartswitch():
     if not platform:
         return False
 
-    # get platform.json file path
-    platform_json = os.path.join(HOST_DEVICE_PATH, platform, "platform.json")
-    try:
-        with open(platform_json, 'r') as f:
-            platform_cfg = json.loads(f.read())
-            return "DPUS" in platform_cfg
-    except IOError:
+    # Retrieve platform.json data
+    platform_data = get_platform_json_data()
+    if platform_data:
+        return "DPUS" in platform_data
+
+    return False
+
+
+def is_dpu():
+    # Get platform
+    platform = get_platform()
+    if not platform:
+        return False
+
+    if not is_smartswitch():
         return False
 
+    # Retrieve platform.json data
+    platform_data = get_platform_json_data()
+    if platform_data:
+        return "DPU" in platform_data
```

Comment:
@vvolam platform.json file on the DPU will have DPU? Is this aligned to smartswitch HLD?

Reply:
@prgeor it is based on https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md#dpu-platformjson

Reply:
Thank you @oleksandrivantsiv